### DISCUSSION TITLE
[Q&A] Network Discovery

### DISCUSSION THREAD

**Discussion by gustavongdp** [glpi-project/glpi-agent#318] [Q&A]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

I am not able to inventory, router, printer, etc..
Inventory of computers, worked perfectly, installed the glpi agent on each computer, but the discovery of network devices, not working. I already created a task, and the network discovery of my router, which is on the ip 192.168.15.1, comes with the type of computer, not as a network device. Any ideas to solve this case?
![image](https://user-images.githubusercontent.com/106355028/213445417-608dea9a-c485-47ec-9a64-54c88bad7525.png)

**Comment by g-bougard** [comment]
As the log is saying, did you associate the right snmp credentials to the ip range ? Also is snmp supported by your device and enabled ?
This message is only displayed when the device is not answering on snmp requests and it only answers to ping requests. It is reported as "Computer" as this is the default type in GLPI. This may change in the future.

**Reply by gustavongdp** [reply]
Sim, adicionei as credenciais na versão 1 e 2, sem sucesso.
Depois entrei nas configurações do roteador, e ativei o snmp v3, adicionei a credencial ao glpi, e sem sucesso de novo.
![image](https://user-images.githubusercontent.com/106355028/213463211-cd000773-f04a-428a-ac7b-a6f20f12d2b1.png)

**Reply by g-bougard** [reply]
Can you please communicate in english here ?

**Reply by gustavongdp** [reply]
Yes, I added credentials in version 1 and 2, to no avail.
Then I entered the router settings, and activated the snmp v3, added the credential to the glpi, and without success again.

**Reply by g-bougard** [reply]
You must also edit the ip range and link the credentials to ip range. An ip range without the expected credentials linked will use only snmp v1 by default.

**Reply by gustavongdp** [reply]
I managed to inventory a printer, but the router without success, even entering the router settings and activating SNMP V3

**Reply by g-bougard** [reply]
Are you able to use [glpi-netinventory](https://glpi-agent.readthedocs.io/en/latest/man/glpi-netinventory.html) script to create an inventory file of your device ?

**Reply by gustavongdp** [reply]
I don't understand friend, could you exemplify, please.

**Reply by g-bougard** [reply]
There are examples at the end of the man page in the doc.
